src/vm/eventpipeevent.cpp

```diff
@@ -156,16 +156,15 @@ unsigned int EventPipeEvent::GetMetadataLength() const
 void EventPipeEvent::RefreshState()
 {
     LIMITED_METHOD_CONTRACT;
-    m_enabled = m_pProvider->EventEnabled(m_keywords, m_level);
+    _ASSERTE(EventPipe::IsLockOwnedByCurrentThread());
+    m_enabledMask = m_pProvider->ComputeEventEnabledMask(m_keywords, m_level);
 }
 
 bool EventPipeEvent::IsEnabled(uint64_t sessionMask) const
 {
     LIMITED_METHOD_CONTRACT;
-    _ASSERT(m_pProvider != nullptr);
-    const bool IsProviderEnabled = m_pProvider->IsEnabled(sessionMask);
-    const bool IsEventEnabled = m_pProvider->EventEnabled(m_keywords, m_level);
-    return (IsProviderEnabled && IsEventEnabled);
+    _ASSERTE(m_pProvider != nullptr);
+    return (m_pProvider->IsEnabled(sessionMask) && (m_enabledMask & sessionMask) != 0);
```

Comment:
Is the m_pProvider check necessary? m_enabledMask already used the provider enabled state when computing the mask bits